app/src/app/api/v0/auth/requestpassword/route.ts

```diff
@@ -0,0 +1,36 @@
+import { db } from "@/models";
+import { apiHandler } from "@/util/api";
+import { passwordRegex, resetPasswordRequest } from "@/util/validation";
+import bcrypt from "bcrypt";
+import createHttpError from "http-errors";
+import jwt from "jsonwebtoken";
+import { Session } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+import { z } from "zod";
+
+const requestPassword = z.object({
+  password: z.string().min(4).regex(passwordRegex),
+});
+
+export const POST = apiHandler(
+  async (
+    _req: NextRequest,
+    context: { session?: Session; params: Record<string, string> },
+  ) => {
+    const body = requestPassword.parse(await _req.json());
+
+    const userId = context.session?.user.id;
+    const user = await db.models.User.findOne({ where: { userId } });
+
+    if (!user) {
+      throw createHttpError.NotFound("User not found!");
+    }
+
+    const password = body.password;
+    const comparePassword = await bcrypt.compare(password, user.passwordHash!);
```

Comment:
I'm not sure if it's a good idea to expose this via an API endpoint. This could very easily lead to brute forcing attacks. For which functionality is this route necessary?